File: examples/langchain/chatbot/agent.py
"""
Minimal chatbot built with LangChain deployed on Superserve.
"""

from langchain_core.messages import HumanMessage
from langchain_openai import ChatOpenAI

llm = ChatOpenAI(model="gpt-4o")

while True:
    try:
        user_input = input()
    except EOFError:
        break
    response = llm.invoke([HumanMessage(content=user_input)])
    print(response.content)

File: examples/typescript/langchain/chatbot/agent.ts
/**
 * Minimal chatbot built with LangChain deployed on Superserve.
 */

import { HumanMessage, SystemMessage } from "@langchain/core/messages"
import { ChatOpenAI } from "@langchain/openai"
import { createInterface } from "readline"

const llm = new ChatOpenAI({ model: "gpt-4o" })
const system = new SystemMessage("You are a helpful assistant.")

const rl = createInterface({ input: process.stdin })
rl.on("line", async (line) => {
  const response = await llm.invoke([system, new HumanMessage(line)])
  console.log(response.content)
})
